gsa/src/web/graphql/wizards.js

```diff
@@ -66,8 +68,108 @@ export const useRunQuickFirstScan = () => {
         });
       });
     },
-    [createTarget, createTask, startTask, sessionTimeout, userTimezone],
+    [createTarget, createTask, startTask],
   );
 
   return [runQuickFirstScan];
 };
+
+export const useRunModifyTask = () => {
+  const [createAlert] = useCreateAlert();
+  const [createSchedule] = useCreateSchedule();
+  const [modifyTask] = useModifyTask();
+
+  const runModifyTask = useCallback(
+    ({
+      alert_email,
+      start_date,
+      start_minute,
+      start_hour,
+      start_timezone,
+      reschedule,
+      tasks,
+    }) => {
+      start_date.hours(start_hour);
+      start_date.minutes(start_minute);
+
+      const [task] = tasks;
+      const {name, id: taskId, alerts} = task;
+
+      const rightNow = new Date().toISOString();
+
+      const event = Event.fromData(
+        {startDate: start_date},
+        start_timezone,
+        false, // if true, then the event will be scheduled at the given time, but at the user timezone instead. If I create an event at 22:00 Shanghai time, it will always be scheduled at 22:00 CEST (User timezone)
+      );
+      let schedulePromise;
+
+      if (reschedule) {
+        const createScheduleInput = {
+          name: `Schedule for ${name} - ${rightNow}`,
+          icalendar: event.toIcalString(),
+          timezone: start_timezone,
+          comment: 'Automatically generated by wizard',
+        };
+        schedulePromise = createSchedule(createScheduleInput);
+      } else {
+        schedulePromise = Promise.resolve();
+      }
+
+      return schedulePromise.then(resp => {
+        // Should be undefined if no schedule is created
+        const scheduleId = resp?.data?.createSchedule?.id;
+
+        let alertPromise;
+
+        if (alert_email.length > 0) {
```

Comment:
Personally I would check if alert email is also defined to not crash if it isn't.
```suggestion
        if (isDefined(alertEmail) && alertEmail.length > 0) {
```